### PR TITLE
feat(loader, loader-bar): describe loader and loader-bar components using design tokens - FE-4452

### PR DESCRIPTION
**Pull request by JoannaMikul**
### Proposed behaviour

Describes the loader and loader-bar components using design tokens.
Updates tests after making the above changes.

### Current behaviour

The loader and loader-bar components use the theme styled-component property.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [x] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required

#### QA

- [ ] Tested in CodeSandbox/storybook
- [ ] Add new Cypress test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions

The correct appearance can be checked in the component: Loader, Loader Bar.
For theme: mint and aegean there should be no differences.
Use devtools to check if loaders use css variables.

### CODE REVIEW

File: src/style/design-tokens/theme-compatibility.util.js [hunk 1]
```diff
@@ -46,7 +47,7 @@ export default (theme) => ({
   colorsUtilityReadOnly400: "#F2F5F6",
   colorsUtilityReadOnly500: "#E6EBED",
   colorsUtilityReadOnly600: "#CCD6DB",
-  colorsActionMajor150: "#B3D9C8",
+  colorsActionMajor150: theme.colors.loadingBarBackground, // TODO (design-tokens): update after merge FE-4443
```
Comment (samtjo) [inline src/style/design-tokens/theme-compatibility.util.js:50]: FE-4443 is merged so this might as well be updated before we merge this one in 👍🏻

**Review by samtjo — CHANGES_REQUESTED** [review]

**Review by jhajduk-sage — DISMISSED** [review]

**Review by samtjo — APPROVED** [review]

File: src/components/loader-bar/loader-bar.style.js [hunk 2]
```diff
@@ -24,20 +24,20 @@ const innerBarAnimation = keyframes`
 `;
 
 const StyledLoaderBar = styled.div`
-  ${({ size, theme }) => css`
+  ${({ size }) => css`
```
Comment (mkrds) [inline src/components/loader-bar/loader-bar.style.js:27]: Remove `import baseTheme from "../../style/themes/base";`

Comment (JoannaMikul) [reply src/components/loader-bar/loader-bar.style.js:27]: StyledLoader uses ${margin} so it's needed.

StyledLoader.defaultProps = {
  theme: baseTheme,
};

Comment (mkrds) [reply src/components/loader-bar/loader-bar.style.js:27]: You're right I didn't notice that

**Review by mkrds — DISMISSED** [review]

**Review by samtjo — DISMISSED** [review]

**Review by mkrds — DISMISSED** [review]

**Review by jhajduk-sage — APPROVED** [review]

**Review by samtjo — APPROVED** [review]

**Comment by sianford** [comment]
Tested in Safari and Chrome in the Chromatic published Storybook. Checked all themes and colours are correct as per the designs and available colour palette and design token themes. New semantic neutral theme to test in a future iteration. Testing passed ✅

**Review by sianford — DISMISSED** [review]

**Review by samtjo — APPROVED** [review]

**Review by mkrds — APPROVED** [review]

**Review by sianford — APPROVED** [review]

**Pull request merged**